yt_dlp/extractor/jiosaavn.py

```diff
@@ -36,29 +36,39 @@ class JioSaavnSongIE(JioSaavnBaseIE):
     def _real_extract(self, url):
         audio_id = self._match_id(url)
         song_data = self._extract_initial_data(url, audio_id)['song']['song']
-        media_data = self._download_json(
-            'https://www.jiosaavn.com/api.php', audio_id, data=urlencode_postdata({
-                '__call': 'song.generateAuthToken',
-                '_format': 'json',
-                'bitrate': '128',
-                'url': song_data['encrypted_media_url'],
-            }))
-
-        return {
-            'id': audio_id,
-            'url': media_data['auth_url'],
-            'ext': media_data.get('type'),
-            'vcodec': 'none',
-            **traverse_obj(song_data, {
-                'title': ('title', 'text'),
-                'album': ('album', 'text'),
-                'thumbnail': ('image', 0, {url_or_none}),
-            }),
-        }
+        formats = []
+        # available bitrates are 16, 32, 64, 128, 320
+        for bitrate in self._configuration_arg('bitrate', ['16', '32', '64', '128', '320'], ie_key='JioSaavn'):
+            media_data = self._download_json(
+                'https://www.jiosaavn.com/api.php', audio_id, f'Downloading format info for {bitrate}',
+                fatal=False, data=urlencode_postdata({
+                    '__call': 'song.generateAuthToken',
+                    '_format': 'json',
+                    'bitrate': bitrate,
+                    'url': song_data['encrypted_media_url'],
+                }))
+            if not media_data.get('auth_url'):
+                self.report_warning(f'Unable to extract format info for {bitrate}')
+                continue
+            formats.append({
+                'url': media_data['auth_url'],
+                'ext': media_data.get('type'),
+                'id': bitrate,
+                'format_id': bitrate,
+                'abr': int(bitrate),
+                'vcodec': 'none',
+                'acodec': media_data.get('type'),
+                **traverse_obj(song_data, {
+                    'title': ('title', 'text'),
+                    'album': ('album', 'text'),
+                    'thumbnail': ('image', 0, {url_or_none}),
+                }),
+            })
+        return self.playlist_result(formats)
```

Comment:
These are formats of a single video and not different videos of a playlist.
We can also add more metadata.
```suggestion
            formats.append({
                'url': media_data['auth_url'],
                'ext': media_data.get('type'),
                'format_id': bitrate,
                'abr': int(bitrate),
                'vcodec': 'none',
            })
        return {
            'id': audio_id,
            'formats': formats,
            **traverse_obj(song_data, {
                'title': ('title', 'text'),
                'album': ('album', 'text'),
                'thumbnail': ('image', 0, {url_or_none}),
                'duration': ('duration', {int_or_none}),
                'view_count': ('play_count', {int_or_none}),
                'release_year': ('year', {int_or_none}),
            }),
        }
```
Tests will need to be updated and `int_or_none` imported from `..utils`.